src/Orleans.Core/Messaging/ILegacyGatewayListProviderConfigurator.cs

```diff
@@ -57,13 +56,28 @@ internal class LegacyGatewayProviderConfigurator
             }
             configurator?.ConfigureServices(clientConfiguration, services);
         }
+
+
+        /// <summary>
+        /// Create instance for type T in certain assembly, using its parameterless constructor
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="assemblyName"></param>
+        /// <returns></returns>
+        internal static T CreateInstanceWithParameterlessConstructor<T>(string assemblyName)
+        {
+            var assembly = Assembly.Load(new AssemblyName(assemblyName));
+            var foundType = TypeUtils.GetTypes(assembly, type => typeof(T).IsAssignableFrom(type), null).FirstOrDefault();
```

Comment:
You removed the `First()`, but you must now check for null explicitly and throw a specific InvalidOperationException if it is. This is particularly useful in case the users use `Custom`.